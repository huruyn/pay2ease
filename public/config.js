window.API_URL = "https://zippy-retail-api-dev.azurewebsites.net";