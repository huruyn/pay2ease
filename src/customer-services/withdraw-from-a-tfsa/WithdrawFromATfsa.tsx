import {
  Row,
  Col,
  Form,
  FormControl,
  Button,
  InputGroup,
} from 'react-bootstrap';

export default function WithdrawFromATfsa() {
  return (
    <div className="page-content">
      <div style={{ padding: 10, backgroundColor: '#F2F3F2' }}>
        <p style={{ textAlign: 'center', marginBottom: 0 }}>
          Withdraw from a TFSA
        </p>
      </div>
      <div style={{ padding: 10, backgroundColor: '#FAF0F3' }}>
        <div className="d-flex justify-content-between">
          <p style={{ marginBottom: 0, fontSize: 14 }} className="d-flex">
            <div
              style={{
                width: 20,
                height: 25,
                border: '1px dotted grey',
                textAlign: 'center',
                marginRight: 10,
              }}
            >
              P
            </div>
            You don&apos;t have an account that allows you to complete this request.
          </p>
          <p>
            {'{'}
            Result #0328
            {'}'}
          </p>
        </div>
      </div>
      <div style={{ padding: 10, backgroundColor: '#DEE9F7' }}>
        <p style={{ marginBottom: 0, fontSize: 14 }} className="d-flex">
          <div
            style={{
              width: 20,
              height: 25,
              border: '1px dotted grey',
              textAlign: 'center',
              marginRight: 10,
            }}
          >
            P
          </div>
          <ul>
            <li>
              Important Information for Regular Investment Plans(RIP): RIP funds
              take 5 business days to clear after your date of deposit. Once
              cleared, you can withdraw the RIP amount.
              <p>
                For example, your monthly RIP of $1,000 is deposited for a total
                TFSA balance of $10,000. Until the $1,000 clears on the 5th
                business day, ZippyX will only process a withdrawal of up to
                $9,000.
              </p>
            </li>
            <li>All fields are required unless stated as optional.</li>
          </ul>
        </p>
      </div>
      <Row className="mt-4">
        <Col md={12} className="bg-light p-3" style={{ fontWeight: 500 }}>
          <span style={{ color: 'rgb(220, 53, 69)', marginRight: 5 }}>+</span>
          <span>Before you begin</span>
        </Col>
      </Row>
      <Row className="mt-3">
        <Col md={6}>
          <Form.Label>From</Form.Label>
          <Form.Select>
            <option> </option>
          </Form.Select>
        </Col>
        <Col md={6}>
          <Form.Label>To</Form.Label>
          <Form.Select>
            <option>- Choose an account -</option>
          </Form.Select>
        </Col>
      </Row>
      <Row className="mt-3">
        <Col md={6}>
          <Form.Label>Amount</Form.Label>
          <InputGroup className="mb-3">
            <InputGroup.Text id="basic-addon1">$</InputGroup.Text>
            <FormControl
              placeholder="Amount"
              aria-label="Amount"
              aria-describedby="basic-addon1"
            />
          </InputGroup>
        </Col>
        <Col md={6}>
          <Form.Label>Date of transfer</Form.Label>
          <Row>
            <Col xs="auto">
              <Form.Select>
                <option>Month</option>
              </Form.Select>
            </Col>
            <Col xs="auto">
              <Form.Select>
                <option>Day</option>
              </Form.Select>
            </Col>
            <Col xs="auto">
              <Form.Select>
                <option>Year</option>
              </Form.Select>
            </Col>
          </Row>
        </Col>
      </Row>
      <Row className="mt-3">
        <Col md={6}>
          <Form.Label htmlFor="email">Email</Form.Label>
          <FormControl id="email" aria-describedby="email" />
          <Form.Check
            type="checkbox"
            label="Send status updates to this email address"
          />
        </Col>
      </Row>
      <Row className="mt-4 pt-3" style={{ borderTop: '1px solid #e7e7e7' }}>
        <Col>
          <Button variant="light" className="d-flex">
            <div
              style={{
                width: 20,
                height: 25,
                border: '1px dotted grey',
                textAlign: 'center',
                marginRight: 10,
              }}
            >
              P
            </div>
            Cancel
          </Button>
        </Col>
        <Col className="d-flex justify-content-end">
          <Button className="zippy-btn d-flex">
            Next
          </Button>
        </Col>
      </Row>
    </div>
  );
}
