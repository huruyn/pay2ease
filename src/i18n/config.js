import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: require('./locales/en-CA/translations.json')
    },
    'en-US': {
      translations: require('./locales/en-US/translations.json')
    },
    'es-US': {
      translations: require('./locales/es-US/translations.json')
    },
    'fr-CA': {
      translations: require('./locales/fr-CA/translations.json')
    },
  },
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;