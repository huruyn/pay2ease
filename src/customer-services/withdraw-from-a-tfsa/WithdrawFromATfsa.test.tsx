import '../../i18n/config';
import WithdrawFromATfsa from './WithdrawFromATfsa';

const ReactTestRenderer = require('react-test-renderer');

describe('WithdrawFromATfsa Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(<WithdrawFromATfsa />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
