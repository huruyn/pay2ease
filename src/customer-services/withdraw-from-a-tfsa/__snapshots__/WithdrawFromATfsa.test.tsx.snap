// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithdrawFromATfsa Component matches the snapshot 1`] = `
<div
  className="page-content"
>
  <div
    style={
      Object {
        "backgroundColor": "#F2F3F2",
        "padding": 10,
      }
    }
  >
    <p
      style={
        Object {
          "marginBottom": 0,
          "textAlign": "center",
        }
      }
    >
      Withdraw from a TFSA
    </p>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#FAF0F3",
        "padding": 10,
      }
    }
  >
    <div
      className="d-flex justify-content-between"
    >
      <p
        className="d-flex"
        style={
          Object {
            "fontSize": 14,
            "marginBottom": 0,
          }
        }
      >
        <div
          style={
            Object {
              "border": "1px dotted grey",
              "height": 25,
              "marginRight": 10,
              "textAlign": "center",
              "width": 20,
            }
          }
        >
          P
        </div>
        You don't have an account that allows you to complete this request.
      </p>
      <p>
        {
        Result #0328
        }
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#DEE9F7",
        "padding": 10,
      }
    }
  >
    <p
      className="d-flex"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 0,
        }
      }
    >
      <div
        style={
          Object {
            "border": "1px dotted grey",
            "height": 25,
            "marginRight": 10,
            "textAlign": "center",
            "width": 20,
          }
        }
      >
        P
      </div>
      <ul>
        <li>
          Important Information for Regular Investment Plans(RIP): RIP funds take 5 business days to clear after your date of deposit. Once cleared, you can withdraw the RIP amount.
          <p>
            For example, your monthly RIP of $1,000 is deposited for a total TFSA balance of $10,000. Until the $1,000 clears on the 5th business day, ZippyX will only process a withdrawal of up to $9,000.
          </p>
        </li>
        <li>
          All fields are required unless stated as optional.
        </li>
      </ul>
    </p>
  </div>
  <div
    className="mt-4 row"
  >
    <div
      className="bg-light p-3 col-md-12"
      style={
        Object {
          "fontWeight": 500,
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgb(220, 53, 69)",
            "marginRight": 5,
          }
        }
      >
        +
      </span>
      <span>
        Before you begin
      </span>
    </div>
  </div>
  <div
    className="mt-3 row"
  >
    <div
      className="col-md-6"
    >
      <label
        className="form-label"
      >
        From
      </label>
      <select
        className="form-select"
      >
        <option>
           
        </option>
      </select>
    </div>
    <div
      className="col-md-6"
    >
      <label
        className="form-label"
      >
        To
      </label>
      <select
        className="form-select"
      >
        <option>
          - Choose an account -
        </option>
      </select>
    </div>
  </div>
  <div
    className="mt-3 row"
  >
    <div
      className="col-md-6"
    >
      <label
        className="form-label"
      >
        Amount
      </label>
      <div
        className="mb-3 input-group"
      >
        <span
          className="input-group-text"
          id="basic-addon1"
        >
          $
        </span>
        <input
          aria-describedby="basic-addon1"
          aria-label="Amount"
          className="form-control"
          placeholder="Amount"
        />
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <label
        className="form-label"
      >
        Date of transfer
      </label>
      <div
        className="row"
      >
        <div
          className="col-auto"
        >
          <select
            className="form-select"
          >
            <option>
              Month
            </option>
          </select>
        </div>
        <div
          className="col-auto"
        >
          <select
            className="form-select"
          >
            <option>
              Day
            </option>
          </select>
        </div>
        <div
          className="col-auto"
        >
          <select
            className="form-select"
          >
            <option>
              Year
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-3 row"
  >
    <div
      className="col-md-6"
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        aria-describedby="email"
        className="form-control"
        id="email"
      />
      <div
        className="form-check"
      >
        <input
          className="form-check-input"
          disabled={false}
          type="checkbox"
        />
        <label
          className="form-check-label"
          title=""
        >
          Send status updates to this email address
        </label>
      </div>
    </div>
  </div>
  <div
    className="mt-4 pt-3 row"
    style={
      Object {
        "borderTop": "1px solid #e7e7e7",
      }
    }
  >
    <div
      className="col"
    >
      <button
        className="d-flex btn btn-light"
        disabled={false}
        type="button"
      >
        <div
          style={
            Object {
              "border": "1px dotted grey",
              "height": 25,
              "marginRight": 10,
              "textAlign": "center",
              "width": 20,
            }
          }
        >
          P
        </div>
        Cancel
      </button>
    </div>
    <div
      className="d-flex justify-content-end col"
    >
      <button
        className="zippy-btn d-flex btn btn-primary"
        disabled={false}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
